src/cmap/worker.rs

```diff
@@ -189,7 +207,17 @@ impl ConnectionPoolWorker {
             .unwrap_or(MAINTENACE_FREQUENCY);
 
         #[cfg(not(test))]
-        let (state, maintenance_frequency) = (PoolState::New, MAINTENACE_FREQUENCY);
+        let (mut state, maintenance_frequency) = (PoolState::New, MAINTENACE_FREQUENCY);
+
+        if options
```

Comment:
we can use `is_load_balanced` here to avoid having to go through the options again.

Reply:
Done.